### DISCUSSION TITLE
PC Speaker not sounding properly under buggy ACPI BIOS

### DISCUSSION THREAD

**Discussion by lah7** [spkr-beep/beep#37] [General]
I might be the one of the few people who still like hearing an audible feedback coming from inside a desktop PC!

Since my motherboard (CPU socket) got damaged months ago, it was time to upgrade. To much dismay, the PC speaker isn't working as expected under this new board/generation. After troubleshooting, I found that passing `acpi=off` to the Linux kernel [parameters](https://www.kernel.org/doc/Documentation/admin-guide/kernel-parameters.txt) makes it work, albeit breaks other stuff too (like GPU, not a good solution). That helps narrow down to being ACPI related, since POST and bootloader beeps (non-ACPI OS stages) are working.

It sounds like a `tsk-tsk` when the beep is supposed to play, and another `tsk-tsk` when the beep is supposed to end.

This is a **MSI PRO Z690-A WIFI DDR4** motherboard with a **Intel Core i5-13600K** CPU. It runs Arch Linux so it has the latest Linux kernel (or LTS). Tried contacting MSI Support and put some video slides together describing the problem for them. Didn't get much help; they concluded it's not a BIOS bug, and I should contact the _Linux official_. 🐧 🤷🏼‍♂️ 

I _suspected_ BIOS bug because the kernel log complains of many `ACPI BIOS Error (bug): Failure creating named object` but they're about XHCI. To see if it was just this board, I tried my father's **ASUS ROG STRIX B760-A GAMING WIFI D4** (with identical CPU) and found it has the same symptoms - but at least doesn't have `ACPI BIOS Error (bug)` in the kernel logs.

So... kind of stumped. Really unsure where in the software/hardware stack things have regressed to feel informed of where to go to get it fixed (and learn something new along the way). If anyone out there can shed knowledge (or a patch!) on whether it's likely the kernel, PIT, DMA/IRQ, BIOS or chipset, that'll be much appreciated! Thanks!

_Or, if you have a LGA 1700 motherboard and `beep` just works, be good to hear from you too!_

**Comment by ndim** [comment]
I fear I cannot be of much help here.

The most recent pieces of Intel hardware I have at home is a late 1996 Pentium MMX and a 1991 486DX33, and both my 2010 era AMD A6 boards (MSI and ASUS) and my 2022 era AMD 5600X board (ASUS) beep just fine.

This looks to me to be a question for someone who actually knows what is supposed to be happening in the hardware, firmware, and kernel, which is not me.

**Comment by lah7** [comment]
Looks like it is really down to the BIOS. Since replacing my board with **ASRock Z790 Nova WiFi**, it works! 🔈 🎶 👍🏼 

Tried other people's boards which had the same buggy symptoms:

* ASUS ROG STRIX B760-A GAMING WIFI D4 
* MSI PRO Z690-A WIFI DDR4 
* MSI Z790 GAMING PLUS WIFI

If anyone stumbles across this in future - you _could_ try contacting the motherboard's technical support and hope it lands in front of an engineer who can understand the problem. 

Otherwise, I didn't really get to the bottom on how (and if) it can be fixed. I saw a feature for the Linux kernel, but ACPI tables look like it requires someone to really know what they're doing:

* [Upgrading ACPI tables via initrd](https://www.kernel.org/doc/html/latest/admin-guide/acpi/initrd_table_override.html)

**Comment by GeniusPilot2016** [comment]
Also, one of the user of my NeoBleeper software was complained about this issue about a month ago, so I researched about this issue and I found out that putting the computer to sleep mode and waking up temporarily fixes this issue, so I added a method to detect such affected chipsets using regular expressions patterns and a method that attempts to reset the PIT and system speaker (aka PC speaker) gate that attempts to mimic sleep mode and waking up if the chipset is affected, but I don't know will it work, because I can't test in my computers, since I only have unaffected computers, whose are an HP 15-af110nt laptop and a desktop computer with Gigabyte B450M H motherboard.
